stages/org.osbuild.uki

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/python3
+"""
+Create a Unified Kernel Image
+
+A Unified Kernel Image (UKI) is an UEFI executable that contains
+an UEFI stub loader, together with the kernel, the initrd, the
+kernel command line and (optionally) a splash screen.
+
+For more information see the specification at:
+https://github.com/uapi-group/specifications/blob/main/specs/unified_kernel_image.md
+
+Host commands: objdump, objcopy
+"""
+
+import os
+import subprocess
+import sys
+from typing import Dict, Tuple
+
+import osbuild
+from osbuild.util import pe32p
+
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "required": ["filename"],
+  "properties": {
+    "filename": {
+      "description": "Filename to use for the resulting UKI",
+      "type": "string"
+    },
+    "kernel": {
+      "type": "object",
+      "required": ["opts"],
+      "properties": {
+        "opts": {
+          "description": "Array of group names for this user",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    }
+  }
+},
+"inputs": {
+  "type": "object",
+  "additionalProperties": false,
+  "required": ["kernel", "initrd", "osrel"],
+  "properties": {
+    "osrel": {
```

Comment:
What's this?

Reply:
OS release information, encoded in the UKI to allow automatic sorting and picking the newest release of a set of UKIs.